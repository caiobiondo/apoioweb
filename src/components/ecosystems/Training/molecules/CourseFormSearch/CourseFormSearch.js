import React from 'react';
import { Paper, FormButton } from 'natura-ui';
import SectionTitle from 'components/molecules/SectionTitle/SectionTitle';
import BaseSearch from '../../../../molecules/BaseSearch';
import { translate } from 'locale';
import PropTypes from 'prop-types';

import {
  Wrapper,
  FormWrapper,
  FormButtonWrapper,
  FormInputWrapper,
  searchButtonStyles,
  FormInput,
  BaseFormSearchDescription,
  FormSelect,
} from './CourseFormSearch.styles';

export class CourseFormSearch extends BaseSearch {
  componentWillMount() {
    const { searchValue, status } = this.props;
    if (searchValue) this.setState({ ...this.state, name: searchValue });
    if (status) this.setState({ ...this.state, status });
  }

  render() {
    const { props } = this;
    const options = [
      { key: 'finished', value: 'finished', label: translate('trainingCourseStatusFinished') },
      { key: 'pending', value: 'pending', label: translate('trainingCourseStatusPending') },
    ];
    return (
      <Paper style={Wrapper}>
        <SectionTitle {...props.sectionTitle} />
        <BaseFormSearchDescription>{translate(props.description)}</BaseFormSearchDescription>
        <FormWrapper>
          <FormInputWrapper>
            <FormInput
              type="text"
              name="inputBaseFormSearch"
              onKeyPress={this.onKeyPress}
              onChange={this.handleNameChange}
              label={translate(props.inputLabel)}
              value={this.state.name}
            />
          </FormInputWrapper>

          <FormInputWrapper>
            <FormSelect
              blankOptionText="Selecione aqui o status do treinamento"
              bordered={true}
              label="Status"
              name="statusFormSearch"
              options={options}
              renderBlank={true}
              onChange={this.handleSelectChange}
              value={this.state.status}
            />
          </FormInputWrapper>
          <FormButtonWrapper>
            <FormButton
              primary
              {...searchButtonStyles}
              type="submit"
              onClick={this.onSubmit}
              label={translate('search')}
            />
          </FormButtonWrapper>
        </FormWrapper>
      </Paper>
    );
  }
}

CourseFormSearch.propTypes = {
  sectionTitle: PropTypes.object.isRequired,
  description: PropTypes.string.isRequired,
  inputLabel: PropTypes.string.isRequired,
  onSearch: PropTypes.func.isRequired,
  searchValue: PropTypes.string,
  status: PropTypes.string,
};

export default CourseFormSearch;
