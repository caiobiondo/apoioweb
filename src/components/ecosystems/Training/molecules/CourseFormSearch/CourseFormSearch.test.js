import React from 'react';
import { shallow } from 'enzyme';
import { CourseFormSearch } from '../CourseFormSearch/CourseFormSearch';
import toJson from 'enzyme-to-json';

const baseFormSearchProps = {
  onSearch: jest.fn(),
  searchValue: '',
  selectedValue: '',
  sectionTitle: { iconName: 'ico_add_customer', value: 'myCustomers' },
  description: 'customersSearchInfo',
  inputLabel: 'customerName',
};

describe('BaseFormSearch Organism', () => {
  it('should render input, select and submit button', () => {
    const result = shallow(<CourseFormSearch {...baseFormSearchProps} />);

    expect(toJson(result)).toMatchSnapshot();
  });

  it("should update state's name with value", () => {
    const expectedName = 'a name';

    const result = shallow(<CourseFormSearch {...baseFormSearchProps} />);
    const instance = result.instance();
    instance.handleNameChange(null, expectedName);

    expect(instance.state.name).toEqual(expectedName);
  });

  it("should update state's status with value", () => {
    const expectedStatus = 'finished';

    const result = shallow(<CourseFormSearch {...baseFormSearchProps} />);
    const instance = result.instance();
    instance.handleSelectChange(null, expectedStatus);

    expect(instance.state.status).toEqual(expectedStatus);
  });

  it('should call onSearch with filters', () => {
    const event = {
      stopPropagation: jest.fn(),
    };
    const name = 'a name';

    const result = shallow(<CourseFormSearch {...baseFormSearchProps} />);
    const instance = result.instance();
    instance.setState({ name });
    instance.onSubmit(event);

    expect(baseFormSearchProps.onSearch).toBeCalledWith({ name });
    expect(event.stopPropagation).toBeCalled();
  });

  it('should call onSearch with filters when enter is pressed', () => {
    const event = {
      key: 'Enter',
    };
    const name = 'a name';

    const result = shallow(<CourseFormSearch {...baseFormSearchProps} />);
    const instance = result.instance();
    instance.setState({ name });
    instance.onKeyPress(event);

    expect(baseFormSearchProps.onSearch).toBeCalledWith({ name });
  });

  it('should not call onSearch with filters when enter is not pressed', () => {
    const onSearch = jest.fn();
    const newBaseFormSearchProps = { ...baseFormSearchProps, onSearch };
    const event = {
      key: 'K',
    };
    const name = 'a name';

    const result = shallow(<CourseFormSearch {...newBaseFormSearchProps} />);
    const instance = result.instance();
    instance.setState({ name });
    instance.onKeyPress(event);

    expect(newBaseFormSearchProps.onSearch).not.toBeCalled();
  });
});
