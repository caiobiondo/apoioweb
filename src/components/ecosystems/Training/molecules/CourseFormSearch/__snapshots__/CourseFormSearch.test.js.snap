// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormSearch Organism should render input, select and submit button 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "padding": 30,
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <SectionTitle
    iconName="ico_add_customer"
    value="myCustomers"
  />
  <CourseFormSearchstyles__BaseFormSearchDescription>
    Preencha o campo abaixo para buscar seus clientes
  </CourseFormSearchstyles__BaseFormSearchDescription>
  <CourseFormSearchstyles__FormWrapper>
    <CourseFormSearchstyles__FormInputWrapper>
      <CourseFormSearchstyles__FormInput
        label="Nome"
        name="inputBaseFormSearch"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
    </CourseFormSearchstyles__FormInputWrapper>
    <CourseFormSearchstyles__FormInputWrapper>
      <CourseFormSearchstyles__FormSelect
        blankOptionText="Selecione aqui o status do treinamento"
        bordered={true}
        label="Status"
        name="statusFormSearch"
        onChange={[Function]}
        options={
          Array [
            Object {
              "key": "finished",
              "label": "ConcluÃ­dos",
              "value": "finished",
            },
            Object {
              "key": "pending",
              "label": "Em andamento",
              "value": "pending",
            },
          ]
        }
        renderBlank={true}
        value=""
      />
    </CourseFormSearchstyles__FormInputWrapper>
    <CourseFormSearchstyles__FormButtonWrapper>
      <FormButton
        backgroundColor="#c9c9c9"
        disabled={false}
        height={54}
        label="Buscar"
        labelColor="#fff"
        labelStyle={
          Object {
            "color": "#fff",
            "fontFamily": "Roboto-Regular",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
        link={null}
        onClick={[Function]}
        primary={true}
        raised={false}
        type="submit"
      />
    </CourseFormSearchstyles__FormButtonWrapper>
  </CourseFormSearchstyles__FormWrapper>
</Paper>
`;
